tests/store/test_sqlalchemy_store.py

```diff
@@ -352,40 +352,50 @@ def test_create_run_with_parent_id(self):
-        self.assertEqual(actual.info.name, run_name)
-        self.assertEqual(actual.info.source_type, expected["source_type"])
-        self.assertEqual(actual.info.source_name, expected["source_name"])
-        self.assertEqual(actual.info.source_version, expected["source_version"])
-        self.assertEqual(actual.info.entry_point_name, expected["entry_point_name"])
-        self.assertEqual(actual.info.start_time, expected["start_time"])
-
-        # Run creation should add two additional tags containing the run name and parent run id.
-        # Check for the existence of these two tags
-        self.assertEqual(
-            actual.data.tags, {MLFLOW_PARENT_RUN_ID: parent_run_id, MLFLOW_RUN_NAME: run_name})
-
-    def test_to_mlflow_entity(self):
-        # Create a run and obtain an MLflow Run entity associated with the new run
-        run = self._run_factory()
+        created_run = self.store.create_run(**expected)
+        fetched_run = self.store.get_run(created_run.info.run_uuid)
+
+        for actual in [created_run, fetched_run]:
+            self.assertEqual(actual.info.experiment_id, experiment_id)
+            self.assertEqual(actual.info.user_id, expected["user_id"])
+            self.assertEqual(actual.info.name, run_name)
+            self.assertEqual(actual.info.source_type, expected["source_type"])
+            self.assertEqual(actual.info.source_name, expected["source_name"])
+            self.assertEqual(actual.info.source_version, expected["source_version"])
+            self.assertEqual(actual.info.entry_point_name, expected["entry_point_name"])
+            self.assertEqual(actual.info.start_time, expected["start_time"])
+
+            # Run creation should add two additional tags containing the run name and parent run id.
+            # Check for the existence of these two tags
+            self.assertEqual(
+                actual.data.tags, {MLFLOW_PARENT_RUN_ID: parent_run_id, MLFLOW_RUN_NAME: run_name})
+
+    def test_to_mlflow_entity_and_proto(self):
+        # Create a run and log metrics, params, tags to the run
+        created_run = self._run_factory()
+        run_uuid = created_run.info.run_uuid
+        self.store.log_metric(
+            run_uuid=run_uuid,
+            metric=entities.Metric(key='my-metric', value=3.4, timestamp=0, step=0))
+        self.store.log_param(run_uuid=run_uuid, param=Param(key='my-param', value='param-val'))
+        self.store.set_tag(run_uuid=run_uuid, tag=RunTag(key='my-tag', value='tag-val'))
+
+        # Verify that we can fetch the run & convert it to proto - Python protobuf bindings
+        # will perform type-checking to ensure all values have the right types
+        run = self.store.get_run(run_uuid)
+        run.to_proto()
 
+        # Verify attributes of the Python run entity
         self.assertIsInstance(run.info, entities.RunInfo)
         self.assertIsInstance(run.data, entities.RunData)
 
-        for key, metric in run.data.metrics.items():
```

Comment:
This logic was actually wrong - `run.data.metrics` is a dict of key -> value, not key -> Metric. Updated this test to actually log metrics/tags/params & verify them, in addition to making sure protobuf conversions didn't throw.

Reply:
Ouch...thanks for fixing this. It was passing because `_run_factory()` does not create any metrics, right?

Reply:
yep exactly